sdk/communication/azure-communication-sms/src/main/java/com/azure/communication/sms/SmsAsyncClient.java

```diff
@@ -53,30 +58,34 @@
      * @param from Number that is sending the message.
      * @param to The recipient's phone number.
      * @param message message to send to recipient.
-     * @param smsOptions set options on the SMS request, like enable delivery report, which sends a report
+     * @param options set options on the SMS request, like enable delivery report, which sends a report
      *                   for this message to the Azure Resource Event Grid.
      * @return response for a successful send Sms request.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<SmsSendResult> send(String from, String to, String message, SmsSendOptions smsOptions) {
-        List<String> recipients = new ArrayList<String>();
-        recipients.add(to);
-        SendMessageRequest request = createSendMessageRequest(from, recipients, message, smsOptions);
+    public Mono<Response<SmsSendResult>> sendWithResponse(String from, String to, String message, SmsSendOptions options) {
+        return sendWithResponse(from, to, message, options, null);
+    }
+
+    Mono<Response<SmsSendResult>> sendWithResponse(String from, String to, String message, SmsSendOptions options, Context context) {
         try {
             Objects.requireNonNull(from, "'from' cannot be null.");
             Objects.requireNonNull(to, "'to' cannot be null.");
-            Mono<Response<SmsSendResponse>> responseMono = withContext(context -> this.smsServiceClient.getSms().sendWithResponseAsync(request, context));
-            Response<SmsSendResponse> response = responseMono.block();
-            SmsSendResponse smsSendResponse = response.getValue();
-
-            List<SmsSendResult> result =  convertSmsResults(smsSendResponse.getValue());
-            if (result.size() == 1) {
-                return Mono.just(result.get(0));
-            } else {
-                return monoError(logger, new NullPointerException("no response"));
-            }
-        } catch (NullPointerException ex) {
-            return monoError(logger, ex);
+            List<String> recipients = Arrays.asList(to);
+            SendMessageRequest request = createSendMessageRequest(from, recipients, message, options);
+            return withContext(contextValue -> {
+                if (context != null) {
+                    contextValue = context;
+                }
+                return smsClient.sendWithResponseAsync(request, contextValue)
+                    .flatMap((Response<SmsSendResponse> response) -> {
+                        List<SmsSendResult> smsSendResults = convertSmsSendResults(response.getValue().getValue());
+                        if (!smsSendResults.isEmpty()) {
+                            return Mono.just(new SimpleResponse<SmsSendResult>(response, smsSendResults.get(0)));
```

Comment:
It seems that we do not throw for single sms. So we should make sure the Troubleshooting section reflect this.

Reply:
Just to clarify, we may get an exception from the Gateway still right? For example, a 401 for unauthorized request. So I should still have the try catch? 

We will be added more details for the case when we get 202 in the response, but the message failed to send because it was either formatted incorrectly or unauthorized.

Reply:
If the request fails all together, the main response code would be a failure. Eg if HMAC signature is not valid we would get a 401. But if the request goes through, still some of the individual SMS messages could fail and that status code will be reflected inside the `SmsSendResponseItem` or whatever we called it.